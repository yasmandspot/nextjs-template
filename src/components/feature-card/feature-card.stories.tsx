import type { Meta, StoryObj } from '@storybook/react';
import { FeatureCard } from './feature-card';
import { CircleHelp } from 'lucide-react';

const meta = {
  title: 'Components/FeatureCard',
  component: FeatureCard,
  parameters: {
    layout: 'centered',
    backgrounds: {
      default: 'light',
      values: [
        { name: 'light', value: '#ffffff' },
        { name: 'dark', value: '#333333' },
      ],
    },
  },
  decorators: [
    (Story) => (
      <div className="max-w-sm p-8">
        <Story />
      </div>
    ),
  ],
  argTypes: {
    icon: {
      description: 'The icon component to display',
      control: 'object',
    },
    title: {
      description: 'The title text of the feature',
      control: 'text',
    },
    description: {
      description: 'The description text of the feature',
      control: 'text',
    },
  },
} satisfies Meta<typeof FeatureCard>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    icon: <CircleHelp className="h-6 w-6" />,
    title: 'Example Feature',
    description: 'This is an example feature card that showcases the component.',
  },
};

export const LongContent: Story = {
  args: {
    icon: <CircleHelp className="h-6 w-6" />,
    title: 'Feature with Long Title That Might Wrap to Multiple Lines',
    description:
      'This is a longer description that demonstrates how the feature card handles more content. It includes multiple sentences to show text wrapping behavior.',
  },
};
