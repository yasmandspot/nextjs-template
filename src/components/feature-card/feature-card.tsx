import { Card, CardContent, CardHeader, CardTitle } from '../ui/card';

interface FeatureCardProps {
  icon: React.ReactNode;
  title: string;
  description: string;
}

/**
 * FeatureCard component that displays a feature with an icon, title, and description.
 *
 * @param {FeatureCardProps} props - The component props.
 * @param {React.ReactNode} props.icon - The icon to display.
 * @param {string} props.title - The title of the feature.
 * @param {string} props.description - The description of the feature.
 * @return {JSX.Element} The rendered feature card.
 */
export function FeatureCard({ icon, title, description }: FeatureCardProps) {
  return (
    <Card>
      <CardHeader>
        <CardTitle className="flex items-center space-x-3">
          <div className="flex h-8 w-8 items-center justify-center">{icon}</div>
          <span>{title}</span>
        </CardTitle>
      </CardHeader>
      <CardContent>
        <p className="text-gray-600">{description}</p>
      </CardContent>
    </Card>
  );
}
