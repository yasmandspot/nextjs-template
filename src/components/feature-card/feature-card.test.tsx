import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { FeatureCard } from './feature-card';
import { CircleHelp } from 'lucide-react';

describe('FeatureCard', () => {
  const defaultProps = {
    icon: <CircleHelp data-testid="icon" className="h-6 w-6" />,
    title: 'Test Feature',
    description: 'Test description',
  };

  it('should match snapshot', () => {
    const { container } = render(<FeatureCard {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should display the title correctly', () => {
    render(<FeatureCard {...defaultProps} />);
    expect(screen.getByText('Test Feature')).toBeInTheDocument();
  });

  it('should display the description correctly', () => {
    render(<FeatureCard {...defaultProps} />);
    expect(screen.getByText('Test description')).toBeInTheDocument();
  });

  it('should render the icon', () => {
    render(<FeatureCard {...defaultProps} />);
    // Check if the icon container is present
    expect(screen.getByTestId('icon')).toBeInTheDocument();
  });

  it('should handle long content appropriately', () => {
    const longProps = {
      icon: <CircleHelp className="h-6 w-6" />,
      title: 'Very Long Feature Title That Might Wrap to Multiple Lines',
      description:
        'This is a very long description that should still be displayed properly within the card component and maintain its layout integrity.',
    };

    render(<FeatureCard {...longProps} />);
    expect(screen.getByText(longProps.title)).toBeInTheDocument();
    expect(screen.getByText(longProps.description)).toBeInTheDocument();
  });
});
