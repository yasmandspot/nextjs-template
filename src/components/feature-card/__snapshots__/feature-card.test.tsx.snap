// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureCard should match snapshot 1`] = `
<div>
  <div
    class="rounded-xl border bg-card text-card-foreground shadow"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <div
        class="font-semibold leading-none tracking-tight flex items-center space-x-3"
      >
        <div
          class="flex h-8 w-8 items-center justify-center"
        >
          <svg
            class="lucide lucide-circle-help h-6 w-6"
            data-testid="icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <path
              d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
            />
            <path
              d="M12 17h.01"
            />
          </svg>
        </div>
        <span>
          Test Feature
        </span>
      </div>
    </div>
    <div
      class="p-6 pt-0"
    >
      <p
        class="text-gray-600"
      >
        Test description
      </p>
    </div>
  </div>
</div>
`;
